brushfire-training/src/main/scala/com/stripe/brushfire/training/Evaluators.scala

```diff
@@ -0,0 +1,49 @@
+package com.stripe.brushfire.training
+
+import com.stripe.brushfire._
+import com.twitter.algebird._
+
+case class ChiSquaredEvaluator[L, W](implicit weightMonoid: Monoid[W], weightDouble: W => Double)
+    extends Evaluator[Map[L, W]] {
+  def trainingError(leaves: Iterable[Map[L,W]]) = {
+    val rows = leaves.filter { _.nonEmpty }
+    if (rows.size > 1) {
+      val n = weightMonoid.sum(rows.flatMap { _.values })
+      val rowTotals = rows.map { row => weightMonoid.sum(row.values) }.toList
+      val columnKeys = rows.flatMap { _.keys }.toList
+      val columnTotals = columnKeys.map { column => column -> weightMonoid.sum(rows.flatMap { _.get(column) }) }.toMap
+      Some((for {
+        column <- columnKeys
+        (row, index) <- rows.zipWithIndex
+      } yield {
+        val observed = row.getOrElse(column, weightMonoid.zero)
+        val expected = (columnTotals(column) * rowTotals(index)) / n
+        val delta = observed - expected
+        (delta * delta) / expected
+      }).sum * -1)
+    } else
+      None
+  }
+}
+
+case class MinWeightEvaluator[L, W: Monoid](minWeight: W => Boolean, wrapped: Evaluator[Map[L, W]])
+    extends Evaluator[Map[L, W]] {
+  def trainingError(leaves: Iterable[Map[L,W]]) = {
+    if (leaves.forall {freq => minWeight(Monoid.sum(freq.values))})
```

Comment:
nit: Mind if we move `leaves.forall {freq => minWeight(Monoid.sum(freq.values))}` out to its own `val`?